admin-dev/themes/new-theme/js/pages/order/view/order-product-renderer.js

```diff
@@ -0,0 +1,192 @@
+    $(OrderViewPageMap.productsTablePaginationPrev).removeClass('disabled');
+    if (numPage === 1) {
+      $(OrderViewPageMap.productsTablePaginationPrev).addClass('disabled');
+    }
+    $(OrderViewPageMap.productsTablePaginationNext).removeClass('disabled');
+    if (numPage === totalPage) {
+      $(OrderViewPageMap.productsTablePaginationNext).addClass('disabled');
+    }
+  }
+
+  paginationAddPage(numPage) {
+    const $tablePagination = $(OrderViewPageMap.productsTablePagination);
+    $tablePagination.data('numPages', numPage);
+    const $linkPagination = $(OrderViewPageMap.productsTablePaginationTemplate).clone();
+    $linkPagination.find('span').attr('data-page', numPage);
+    $linkPagination.find('span').html(numPage);
+    $(OrderViewPageMap.productsTablePaginationTemplate).before($linkPagination.removeClass('d-none'));
+  }
+
+  paginationRemovePage(numPage) {
+    const $tablePagination = $(OrderViewPageMap.productsTablePagination);
+    const numPages = $tablePagination.data('numPages');
+    $tablePagination.data('numPages', numPages - 1);
+    $(OrderViewPageMap.productsTablePagination).find(`li:has(> [data-page="${numPage}"])`).remove();
+  }
+
+  paginateRowCreate(result, orderId) {
+    const $productRow = $(OrderViewPageMap.productTemplateRow).clone();
+    $productRow.attr('id', `orderProduct_${result.orderDetailId}`);
+    if (result.imagePath) {
+      $productRow.find('td.cellProductImg img').attr('src', result.imagePath);
+      $productRow.find('td.cellProductImg img').attr('alt', result.name);
+    } else {
+      $productRow.find('td.cellProductImg img').remove();
+    }
+    $productRow.find('td.cellProductName a').attr('href', this.router.generate('admin_product_form', {orderId: result.id}));
+    $productRow.find('td.cellProductName p.productName').html(result.name);
+    if (result.supplierReference) {
+      $productRow.find('td.cellProductName p.productSupplierReference').append(result.supplierReference);
+    } else {
+      $productRow.find('td.cellProductName p.productSupplierReference').remove();
+    }
+    if (result.reference) {
+      $productRow.find('td.cellProductName p.productReference').append(result.reference);
+    } else {
+      $productRow.find('td.cellProductName p.productReference').remove();
+    }
+    $productRow.find('td.cellProductUnitPrice').html(result.unitPrice);
+    if (result.quantity > 1) {
+      $productRow.find('td.cellProductQuantity span.badge').html(result.quantity);
+    } else {
+      $productRow.find('td.cellProductQuantity span.badge').replaceWith(result.quantity);
+    }
+    $productRow.find('td.cellProductQuantity input').val(result.quantity);
+    $productRow.find('td.cellProductLocation').html(result.location);
+    $productRow.find('td.cellProductAvailableQuantity').html(result.availableQuantity);
+    $productRow.find('td.cellProductTotalPrice').html(result.totalPrice);
```

Comment:
You must refactor all `$productRow.find` and group them in variables for performance.